### PR TITLE
Custom lint tests

### PR DESCRIPTION
**Pull request by KamilSztandur**

### CODE REVIEW

File: packages/leancode_lint/test/lints_test_app/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    sdk: flutter
+
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  leancode_lint:
+    path: ../../../leancode_lint
+
+flutter:
+  uses-material-design: true
```
Comment (PiotrRogulski) [inline packages/leancode_lint/test/lints_test_app/pubspec.yaml:22]: Add a final newline

File: packages/leancode_lint/test/lints_test_app/analysis_options.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+          from_package: flutter
+        - instead_of: RichText
+          from_package: flutter
+      LftScaffold:
+        - instead_of: Scaffold
+          from_package: flutter
+
+analyzer:
+  plugins:
+    - custom_lint
```
Comment (PiotrRogulski) [inline packages/leancode_lint/test/lints_test_app/analysis_options.yaml:17]: Ditto

**Review by PiotrRogulski — APPROVED** [review]

File: packages/leancode_lint/test/lints_test_app/lib/use_design_system_item_test.dart
Comment (shilangyu) [inline packages/leancode_lint/test/lints_test_app/lib/use_design_system_item_test.dart:1]: add test case for merely referencing a forbidden type. For example:


```dart
void func(
	Text text,
) {}
```

`Text` here should be linted

File: packages/leancode_lint/test/lints_test_app/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+name: lints_test_app
+description: Simple Flutter app for lint rules testing
```
Comment (shilangyu) [inline packages/leancode_lint/test/lints_test_app/pubspec.yaml:2]:
```suggestion
description: Flutter app for lint rules testing
```

File: packages/leancode_lint/test/lints_test_app/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+environment:
+  sdk: '>=3.1.0 <4.0.0'
+
+dependencies:
+  flutter:
+    sdk: flutter
+
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
```
Comment (shilangyu) [inline packages/leancode_lint/test/lints_test_app/pubspec.yaml:17]: not needed

File: packages/leancode_lint/README.md
Comment (shilangyu) [inline packages/leancode_lint/README.md:1]: revert all changes

File: .github/workflows/leancode_lint-test.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+      - name: Cache pub dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.FLUTTER_HOME }}/.pub-cache
+          key: ${{ runner.os }}-pub-${{ hashFiles('**/pubspec.lock') }}
+          restore-keys: ${{ runner.os }}-pub-
+
+      - name: Run lint rules tests
+        run: cd test/lints_test_app/ && flutter pub get && dart run custom_lint
```
Comment (shilangyu) [inline .github/workflows/leancode_lint-test.yml:50]:
```suggestion
      - name: Run lint rules tests
        working-directory: test/lints_test_app
        run: |
          flutter pub get
          dart run custom_lint
```

File: packages/leancode_lint/test/lints_test_app/lib/add_sliver_prefix_for_widget_returning_sliver_test.dart [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+class NotPrefixedWidgetReturningSliver extends StatelessWidget {
+  const NotPrefixedWidgetReturningSliver({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const SliverToBoxAdapter();
+  }
+}
+
+// Should not throw warning since it's prefixed
```
Comment (shilangyu) [inline packages/leancode_lint/test/lints_test_app/lib/add_sliver_prefix_for_widget_returning_sliver_test.dart:43]: For the future: warnings arent thrown. Either reported or raised.

**Review by shilangyu — CHANGES_REQUESTED** [review]
.dart_tool is in .gitignore, I am not sure how you pushed it

**Review by shilangyu — APPROVED** [review]

**Pull request merged**